edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/BridgeConnection/MqttBridgeConnector.cs

```diff
@@ -0,0 +1,444 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using uPLibrary.Networking.M2Mqtt;
```

Comment:
Couple of questions - 
1. How did you choose this lib? 
2. Any licensing concerns?

Reply:
We should hide all the client related details behind interfaces, so that we can swap it out for a different lib at a later point, if needed.